src/App.tsx

```diff
@@ -1,26 +1,257 @@
+        })
+        .finally(() => {
+          formInput.current?.focus();
+        });
+    });
+  };
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-loading-todos#react-todo-app-load-todos">
-          React Todo App - Load Todos
-        </a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <header className="todoapp__header">
+          {/* this button should have `active` class only if all todos are completed */}
+          <button
+            type="button"
+            className="todoapp__toggle-all active"
+            data-cy="ToggleAllButton"
+          />
+
+          {/* Add a todo on form submit */}
+          <form onSubmit={handleSubmit}>
+            <input
+              data-cy="NewTodoField"
+              type="text"
+              className="todoapp__new-todo"
+              placeholder="What needs to be done?"
+              ref={formInput}
+              value={newTodoTitle}
+              onChange={event => setNewTodoTitle(event.target.value)}
+              disabled={isLoading}
+            />
+          </form>
+        </header>
+
+        {todos.length !== 0 && (
+          <TodoList todos={filteredTodos} onDelete={handleDeleteTodo} />
+        )}
+
+        {tempTodo && (
+          <section className="todoapp__main">
+            <TodoCard todo={tempTodo} onDelete={handleDeleteTodo} />
+          </section>
+        )}
+
+        {/* Hide the footer if there are no todos */}
+        {todos.length !== 0 && (
+          <footer className="todoapp__footer" data-cy="Footer">
+            <span className="todo-count" data-cy="TodosCounter">
+              {todosCounter} items left
+            </span>
+
+            {/* Active link should have the 'selected' class */}
+            <nav className="filter" data-cy="Filter">
+              <a
+                href="#/"
+                className={classNames('filter__link', {
+                  selected: filter === 'all',
+                })}
+                data-cy="FilterLinkAll"
+                onClick={() => onFilterChange('all')}
+              >
+                All
+              </a>
+
+              <a
+                href="#/active"
+                className={classNames('filter__link', {
+                  selected: filter === 'active',
+                })}
+                data-cy="FilterLinkActive"
+                onClick={() => onFilterChange('active')}
+              >
+                Active
+              </a>
+
+              <a
+                href="#/completed"
+                className={classNames('filter__link', {
```

Comment:
Use `Object.values(your created enum)` and render these options with `map()` method